### PR TITLE
Feature/@oa 16 create auction

### PR DESCRIPTION
**Pull request by sawabond**
- Added auction entity
- Added migration with EF Core code first
- Added logic and controller to create an auction
- Added unit tests

`<gateway_url>/auction` POST
```
{
  "name": "Example Auction",
  "description": "This is an example auction.",
  "startTime": "2023-07-15T10:00:00Z",
  "auctionType": 0 // English
}
```

### CODE REVIEW

File: src/api/AuthServer/AuctionService.Application.UnitTests/AuctionServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+using AuctionService.Application.Models.Auction;
+using AuctionService.Core.Entities;
+using AuctionService.Core.Repositories;
+using Microsoft.Extensions.Logging;
+using NSubstitute;
+using AuctionService = AuctionService.Application.Services.AuctionService;
```
Comment (Yugmund) [inline src/api/AuthServer/AuctionService.Application.UnitTests/AuctionServiceTests.cs:6]: Unused AuctionService

File: src/api/AuthServer/AuctionService.Infrastructure/Migrations/20230717123057_AddedAuction.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+﻿using System;
```
Comment (Yugmund) [inline src/api/AuthServer/AuctionService.Infrastructure/Migrations/20230717123057_AddedAuction.cs:1]: Unused System

File: src/api/AuthServer/AuctionService.Infrastructure/Migrations/AuctionDbContextModelSnapshot.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
+﻿// <auto-generated />
+using System;
+using AuctionService.Infrastructure.Persistence;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Metadata;
+using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
```
Comment (Yugmund) [inline src/api/AuthServer/AuctionService.Infrastructure/Migrations/AuctionDbContextModelSnapshot.cs:7]: Unused usings

File: src/api/AuthServer/AuctionService/Program.cs [hunk 4]
```diff
@@ -1,8 +1,12 @@
+using AuctionService.Application.Models.Auction.Validators;
+using AuctionService.Extensions;
 using AuctionService.Hubs;
-using Microsoft.AspNetCore.Mvc;
+using AuctionService.Infrastructure.Persistence;
+using FluentValidation;
+using FluentValidation.AspNetCore;
+using Microsoft.EntityFrameworkCore;
```
Comment (Yugmund) [inline src/api/AuthServer/AuctionService/Program.cs]: Unused usings

**Review by Yugmund — COMMENTED** [review]
Remove unused usings and we can merge. Works OK

File: src/api/AuthServer/AuctionService.Application.UnitTests/AuctionServiceTests.cs (hunk 1 above)
Comment (sawabond) [reply src/api/AuthServer/AuctionService.Application.UnitTests/AuctionServiceTests.cs:6]: Removed unused using directives

File: src/api/AuthServer/AuctionService.Infrastructure/Migrations/20230717123057_AddedAuction.cs (hunk 2 above)
Comment (sawabond) [reply src/api/AuthServer/AuctionService.Infrastructure/Migrations/20230717123057_AddedAuction.cs:1]: Removed unused using directives

File: src/api/AuthServer/AuctionService/Program.cs (hunk 4 above)
Comment (sawabond) [reply src/api/AuthServer/AuctionService/Program.cs]: Removed unused using directives

File: src/api/AuthServer/AuctionService.Infrastructure/Migrations/AuctionDbContextModelSnapshot.cs (hunk 3 above)
Comment (sawabond) [reply src/api/AuthServer/AuctionService.Infrastructure/Migrations/AuctionDbContextModelSnapshot.cs:7]: Removed unused using directives

**Pull request merged**